internal/storage/bucket_handle_test.go

```diff
@@ -1187,3 +1192,26 @@ func (testSuite *BucketHandleTest) TestComposeObjectMethodWithOneSrcObjectIsDstO
 	assert.NotNil(testSuite.T(), composedObj)
 	assert.Equal(testSuite.T(), len(ContentInTestObject)+len(ContentInTestSubObject), int(composedObj.Size))
 }
+
+func (testSuite *BucketHandleTest) TestFetchBucketTypeForHierarchicalNameSpace() {
+	// Set the expectation for GetStorageLayout.
+	testSuite.mockClient.On("GetStorageLayout", mock.Anything, mock.Anything, mock.Anything).
+		Return(&controlpb.StorageLayout{
+			HierarchicalNamespace: &controlpb.StorageLayout_HierarchicalNamespace{Enabled: true},
+		}, nil)
+
+	bucketType := mockFetchAndSetBucketType(testSuite.mockClient)
+
+	assert.Equal(testSuite.T(), gcs.Hierarchical, bucketType, "Expected Hierarchical bucket type")
+}
+
+func (testSuite *BucketHandleTest) TestFetchBucketTypeWithError() {
+	var x *controlpb.StorageLayout
+	// Test when the client returns an error.
+	testSuite.mockClient.On("GetStorageLayout", mock.Anything, mock.Anything, mock.Anything).
+		Return(x, errors.New("mocked error"))
+
+	bucketType := mockFetchAndSetBucketType(testSuite.mockClient)
+
+	assert.Equal(testSuite.T(), gcs.Unknown, bucketType, "Expected Unknown when there's an error")
+}
```

Comment:
These two tests are testing mockFetchAndSetBucketType, actual FetchAndSetBucketType method which will be used in production code is not being tested here. To validate FetchAndSetBucketType try to mock only client and create new bucket hadnle with embedded mock client and in the test actual FetchAndSetBucketType method should be called. 

Also, two more test cases needs to be added here, one when control client is nil and second when storagelayout does not give any error but hierarchial block is not present in success response.

Reply:
Done please check.